Orange/tests/test_table.py

```diff
@@ -53,285 +54,146 @@ def test_filename(self):
-            self.assertEqual(d[0, "b"], 42)
-            d[0]["b"] = 0
-            self.assertEqual(d[0, "b"], 0)
-
-            d[0, varb] = 42
-            self.assertEqual(d[0, "b"], 42)
-            d[0][varb] = 0
-            self.assertEqual(d[0, "b"], 0)
-
-            d[0, 0] = 42
-            self.assertEqual(d[0, "b"], 42)
-            d[0][0] = 0
-            self.assertEqual(d[0, "b"], 0)
-
-            d[0, np.int_(0)] = 42
-            self.assertEqual(d[0, "b"], 42)
-            d[0][np.int_(0)] = 0
             self.assertEqual(d[0, "b"], 0)
+            for pos in ("b", varb, 0, np.int_(0)):
+                d[0, pos] = 42
+                self.assertEqual(d[0, "b"], 42)
+                d[0][pos] = 0
+                self.assertEqual(d[0, "b"], 0)
 
     def test_indexing_del_example(self):
-        import warnings
-
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
+        with catch_warnings():
+            simplefilter("ignore")
             d = data.Table("test2")
             initlen = len(d)
 
-            # remove first
             d[4, "e"] = "4ex"
-            self.assertEqual(d[4, "e"], "4ex")
-            del d[0]
-            self.assertEqual(len(d), initlen - 1)
-            self.assertEqual(d[3, "e"], "4ex")
-
-            # remove middle
-            del d[2]
-            self.assertEqual(len(d), initlen - 2)
-            self.assertEqual(d[2, "e"], "4ex")
-
-            # remove middle
-            del d[4]
-            self.assertEqual(len(d), initlen - 3)
-            self.assertEqual(d[2, "e"], "4ex")
-
             # remove last
             d[-1, "e"] = "was last"
-            del d[-1]
-            self.assertEqual(len(d), initlen - 4)
-            self.assertEqual(d[2, "e"], "4ex")
-            self.assertNotEqual(d[-1, "e"], "was last")
-
-            # remove one before last
-            d[-1, "e"] = "was last"
-            del d[-2]
-            self.assertEqual(len(d), initlen - 5)
-            self.assertEqual(d[2, "e"], "4ex")
-            self.assertEqual(d[-1, "e"], "was last")
+            self.assertEqual(d[4, "e"], "4ex")
+            for index, (delIndex, pos) in zip(range(1, 6), ((0, 3),  # remove first
+                                                            (2, 2),  # remove middle
+                                                            (4, 2),  # remove middle
+                                                            (-1, 2),  # remove last
+                                                            (-2, 2))):  # remove one before last
+                del d[delIndex]
+                self.assertEqual(len(d), initlen - index)
+                self.assertEqual(d[pos, "e"], "4ex")
+                if index is 4:
+                    self.assertNotEqual(d[-1, "e"], "was last")
+                    d[-1, "e"] = "was last"
+                elif index is 5:
+                    self.assertEqual(d[-1, "e"], "was last")
```

Comment:
This refactoring is not ok. Loops are only clearer when there aren't countless obfuscating exceptions breaking their flow.